purchase_batch_invoicing_advanced/models/purchase_batch_invoicing.py

```diff
@@ -0,0 +1,27 @@
+# © 2021 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import models, _
+from odoo.exceptions import UserError
+
+
+class PurchaseBatchInvoicing(models.TransientModel):
+    _inherit = 'purchase.batch_invoicing'
+
+    def grouped_purchase_orders(self):
+        res = super().grouped_purchase_orders()
+        for record in res:
+            if len(record.mapped('order_type.journal_id')) > 1:
+                raise UserError(_('Selected purchase orders contain different billing journals.'))
```

Comment:
Better understanding IMO:
```suggestion
                raise UserError(_('Selected purchase orders within the same group contain different billing journals.'))
```